### ISSUE TITLE
Enable asset checks for a subset of dbt tests

### ISSUE THREAD

**Issue by johannkm** [dagster-io/dagster#18469]
Checks are currently all or nothing with the dbt integration.

> The one feature we miss is the ability to select a set of dbt tests that are loaded as asset checks. Currently, all of our dbt tests are converted to asset checks. We would really benefit from loading only a selected ones, let's say matching some tag. Something along these lines that you already support.

  ```python
    # Select the dbt assets that have the tag "foo".
    foo_selection = build_dbt_asset_selection([dbt_assets], dbt_select="tag:foo"
  ```

>  I would imaging another setting property in DagsterDbtTranslatorSettings

**Comment by mgierada** [comment]
That is definitely something that would make our pipelines shine! Let's imagine that some of the dbt tests are not really crucial or those tests are a relationship tests that takes a lot of time to complete. We might not necessary want them to be loaded as asset checks and be executed shortly after dbt asset materialization. 

An alternative solution could be launching asset checks as a separate jobs that runs in parallel to dbt asset materialization, meaning they are not part of the same job. That would ensure that asset materialization is not delayed by checks. Check can run in parallel. There is already some capacity for that. See `@asset_checks` or `@asset`  decorators and how asset checks are defined there. I would love to port those APIs to `@dbt_asset` decorator.

**Comment by johannkm** [comment]
@mgierada re: 

> An alternative solution could be launching asset checks as a separate jobs that runs in parallel to dbt asset materialization, meaning they are not part of the same job. That would ensure that asset materialization is not delayed by checks

You can do this today with dbt assets. You can build two jobs like this:

```python
assets_only = define_asset_job(
    "my_asset_only_job",
    selection=AssetSelection.assets(my_dbt_assets).without_checks(),
)

checks_only = define_asset_job(
    "checks_only_job", selection=AssetSelection.checks_for_assets(my_dbt_asset)
)
```

Here's the docs: https://docs.dagster.io/concepts/assets/asset-checks#via-sensors-and-schedules

**Comment by geoHeil** [comment]
I think it got better recently https://github.com/dagster-io/dagster/discussions/18957 for `build_dbt_asset_selection` but it is still unclear for me how to use this for the  `dbt_assets` API.

**Comment by NicolasGuary** [comment]
Up on this one, especially as unit tests are being introduced in dbt, it's not really useful to have them loaded as asset checks and so it would be great to exclude tests with a specific tag (and not only having the possibility to select all checks for a given asset).

**Comment by NicolasGuary** [comment]
I think a good idea could be to include this in the `meta` key, for instance in my case, I would use : 

```
tests:
  common:
    unit:
      +meta:
        dagster:
          asset_check_enabled: false
```

Of course, this can be set individually to each dbt test. Only potential issue of this approach is that meta can't be set for generic tests, but should not be an issue (https://docs.getdbt.com/reference/resource-configs/meta)
